projects/ngx-sub-form/src/lib/ngx-sub-form.ts

```diff
@@ -148,22 +148,19 @@ export function createForm<ControlInterface, FormInterface>(
       if (!isRoot<ControlInterface, FormInterface>(options)) {
         return formGroup.valueChanges.pipe(delay(0));
       } else {
-        if (options.manualSave$) {
-          return options.manualSave$.pipe(
-            withLatestFrom(formGroup.valueChanges),
-            map(([_, formValue]) => formValue),
-            filter(() => formGroup.valid),
-            delay(0),
-            filter(formValue => !isEqual(transformedValue, formValue)),
-          );
-        } else {
-          return formGroup.valueChanges.pipe(
-            filter(() => formGroup.valid),
-            delay(0),
-            filter(formValue => !isEqual(transformedValue, formValue)),
-            options.handleEmissionRate ?? identity,
-          );
-        }
+        const formValues$ = options.manualSave$
+          ? options.manualSave$.pipe(
+              withLatestFrom(formGroup.valueChanges),
+              map(([_, formValue]) => formValue),
+            )
+          : formGroup.valueChanges;
+
+        return formValues$.pipe(
+          filter(() => formGroup.valid),
+          delay(0),
+          filter(formValue => (options.outputFilterPredicate ?? isEqual)(transformedValue, formValue)),
+          options.handleEmissionRate ?? identity,
```

Comment:
this was largely a tidyup, but also I realised that it does make sense to be able to do emission rate handling for manual save too, as you might want to limit how often a value is published if the manual save is bound to an on-submit event too. or some other weird and wonderful use-case. It doesn't seem invalid so for simplicity it is better to just permit it imo